indico/modules/events/timetable/templates/pdf/_program.html

```diff
@@ -0,0 +1,272 @@
+                        <h4>
+                            <span class="timebox inherited-bg-color inherited-txt-color">
+                                {# {{ subchild.start_dt|format_time(timezone=entry.event.tz) }} - {{ subchild.end_dt|format_time(timezone=entry.event.tz) }} #}
+                                {% if show_length_contribs %}
+                                    {{ (subchild.duration)|format_human_timedelta(narrow=True) }}
+                                {% endif %}
+                            </span>
+                            {{ subchild.title }}
+                        </h4>
+                        <div class="contrib-timeline-content">
+                            {{ render_nested_contrib(subchild, show_title, show_affiliation, show_length_contribs) }}
+                        </div>
+                    </li>
+                {% endfor%}
+            </ul>
+        </div>
+    {% endif %}
+{% endmacro %}
+
+{% macro render_block_date(entry, day, print_date_close_to_sessions) %}
+    {% if print_date_close_to_sessions %}
+        <p class="session-block-date">
+            <b>{{ day|format_skeleton('EEEEdMMMM') }}</b>
+            <span class="timebox">
+                {{ entry.start_dt|format_time(timezone=entry.event.tz) }} - {{ entry.end_dt|format_time(timezone=entry.event.tz) }}
+            </span>
+        </p>
+    {% endif %}
+{% endmacro %}
+
+{% macro render_time(entry) %}
+    <div class="time">
+        <div class="time-from">
+            {{ entry.start_dt|format_time(timezone=entry.event.tz) }}
+        </div>
+        <div class="time-divider">
+            {# Line between generated in CSS #}
+        </div>
+        <div class="time-divider">
+            {# Line between generated in CSS #}
+        </div>
+        <div class="time-to">
+            {{ entry.end_dt|format_time(timezone=entry.event.tz) }}
+        </div>
+    </div>
+{% endmacro %}
+
+{# Utilities #}
+
+{% macro render_place(entry) %}
+    <b>{% trans %}Location{% endtrans %}:</b>
+    {% if entry.object.venue_name %}
+        {{ entry.object.venue_name }}{% if entry.object.room_name %},{% endif %}
+    {% endif %}
+    {% if entry.object.room_name %}
+        {{ entry.object.room_name }}
+    {%endif%}
+{% endmacro %}
+
+{% macro render_names(people, show_title=false, show_affiliation=false) %}
+    {% for person in people %}
+        {{
+            person.get_full_name(show_title=show_title, abbrev_first_name=false, last_name_first=false, last_name_upper=false) +
+            (' (' + person.affiliation + ')' if show_affiliation and person.affiliation else '') +
+            (',' if not loop.last else '')
+        }}
+    {% endfor %}
+{% endmacro %}
+
+{% macro get_custom_id(entry) -%}
+    {{ 'c' ~ entry.type ~ '-' ~ entry.id }}
+{% endmacro %}
+
+{% macro get_color_styles(entry, colors, transhex='00') %}
+    {% set c_id = get_custom_id(entry) %}
+
+    <style>
+        {{ '.' ~ c_id  ~ ' .inherited-bg-color' }} {
```

Comment:
Isn't this more readable?

```suggestion
        .{{ c_id }} .inherited-bg-color' {
```

Reply:
Surprisingly this makes the code not work. We had this before and had to change the approach that we currently have

Reply:
...how? My typo (extra `'` aside), this should literally generate the same output...

```
>>> Template("{{ '.' ~ c_id  ~ ' .inherited-bg-color' }} {").render(c_id=123)
'.123 .inherited-bg-color {'

>>> Template(".{{ c_id }} .inherited-bg-color {").render(c_id=123)
'.123 .inherited-bg-color {'
```